pkg/scheduler/api/topology_info/topology_info.go

```diff
@@ -3,8 +3,38 @@
 
 package topology_info
 
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/api/common_info"
+)
+
 type TopologyConstraintInfo struct {
 	PreferredLevel string
 	RequiredLevel  string
 	Topology       string
+
+	schedulingConstraintsSignature common_info.SchedulingConstraintsSignature
+}
+
+func (tc *TopologyConstraintInfo) GetSchedulingConstraintsSignature() common_info.SchedulingConstraintsSignature {
+	if tc == nil {
+		return ""
+	}
+
+	if tc.schedulingConstraintsSignature == "" {
+		tc.schedulingConstraintsSignature = tc.generateSchedulingConstraintsSignature()
+	}
+
+	return tc.schedulingConstraintsSignature
+}
+
+func (tc *TopologyConstraintInfo) generateSchedulingConstraintsSignature() common_info.SchedulingConstraintsSignature {
+	hash := sha256.New()
+	hash.Write([]byte(tc.Topology))
+	hash.Write([]byte(tc.RequiredLevel))
```

Comment:
Make sure that required and preferred result in different signature: this looks like it will serialize the same for both